xml/README.md

```diff
@@ -21,9 +21,14 @@
   - Custom `reviver` and `replacer` functions
 - Metadata stored into non-enumerable properties for advanced usage
 
+## Public API
+
+If it's exported, it's public. It's undefined behavior territory if a TypeScript check warns about issues. Don't write to properties starting with a tilde (`~`).
```

Comment:
I'm not sure this is needed

You cannot import anything that's not `export`-ed anyways (and everything public is already documented on either jsr docs / deno.land/x).
As for `~`/`#` properties they're already typed as `readonly` so TS should complain if user tries to write on them, and at runtime it'll error too since they're set as `writable: false` so it doesn't feel necessary to reiterate about it.

Not sure what the second sentence is about, any example of when a check warn is issued ?

Reply:
I'd just been creating objects using ~name and ~children, rather than the actual api. I figured some note might help future readers. On reread, it could certainly use some rephrasing, and it really belongs in a doc comment anyway.
Maybe order of operations would help? A (TypeScript check) warns. Maybe better as when it emits a warning? Anyway, I guess I'll remove this tomorrow. I just wanted it fill it out enough it felt like a whole paragraph, but if it doesn't need to be here, words begone!

Reply:
I guess it's kind of true for `stringify()` but even then technically it's not forbidden to do so (since you can `stringify(parse())` and it'd still be valid despite having `~` properties). These are still part of the public API it's just that they're non-enumerable by default and some of these options makes less sense in `stringify` than in `parse`

I think they should both be handled in code, with `~name` having a check to ensure it matches the name in parent if defined (rather than being ignored currently), and for `~children` it's eventually planned to be supported, it's just it's not implemented yet because of lack of time and it's also a pretty niche usage so not high in the priority list

Anyways apart from this nitpick, the remaining docs looks good to me, so I have no additional changes to request once removed

Reply:
Oh, and there's a line about it anyway:
https://github.com/lowlighter/libs/blob/0e53714b0179c717d44b2b9b3efb3250074b4360/xml/README.md?plain=1#L237-L238